src/org/labkey/snd/SNDUserSchema.java

```diff
@@ -136,15 +136,19 @@ public TableInfo createTable(SNDUserSchema schema, ContainerFilter cf)
                     @Override
                     public TableInfo createTable(SNDUserSchema schema, ContainerFilter cf)
                     {
-                        return new EventsTable(schema, SNDSchema.getInstance().getTableInfoEvents(), cf).init();
+                        if (schema.getContainer().hasPermission(schema.getUser(), SNDViewerPermission.class, schema.getContextualRoles()))
+                        {
+                            return new EventsTable(schema, SNDSchema.getInstance().getTableInfoEvents(), cf).init();
+                        }
+                        return null;
```

Comment:
Can i please know how we are handleing this null

Reply:
in the SNDManager.java starting at line 184:
`   /**
     * Generic get table info function
     */
    public TableInfo getTableInfo(@NotNull UserSchema schema, @NotNull String table)
    {
        TableInfo tableInfo = schema.getTable(table);
        if (tableInfo == null)
            throw new IllegalStateException(table + " TableInfo not found");

        return tableInfo;
    }
`

Reply:
Thanks Terry but in this scenario we should be giving error message related to View/Admin permission. Is it still ok if we throw "TableInfo not found" for the user?

Reply:
Yes. It is a generic function returning a generic message that is technically correct. The UI should prevent the user from trying to access the data if they don't have permission to do so. Jay is working on that code now in snprc_ehr. This permission check will keep them from getting to the data if they follow some other path such as navigating to the schema browser and selecting one of the tables.